legate/core/context.py

```diff
@@ -66,6 +66,35 @@ def find_last_user_frame(libname: str) -> str:
     return f"{frame.f_code.co_filename}:{frame.f_lineno}"
 
 
+class LibraryAnnotations:
+    def __init__(self) -> None:
+        self._entries: dict[str, str] = {}
+
+    @property
+    def empty(self) -> bool:
+        return len(self._entries) == 0
+
+    def __getitem__(self, key: str) -> Optional[str]:
+        return self._entries[key] if key in self._entries else None
```

Comment:
also an option:
```suggestion
        return self._entries.get(key, None)
```

Reply:
https://github.com/nv-legate/legate.core/pull/464/commits/828d27955457b8038419ef37e13dac281118a7ac

Reply:
`__get_item__` and `__set_item__` are removed